src/gtk/toga_gtk/widgets/button.py

```diff
@@ -17,11 +17,18 @@ def set_label(self, label):
         self.rehint()
 
     def set_enabled(self, value):
-        # self._impl.set_sensitive(value)
-        self.interface.factory.not_implemented('Button.set_enabled()')
+        self.native.set_sensitive(value)
 
     def set_background_color(self, value):
-        self.interface.factory.not_implemented('Button.set_background_color()')
+        if value:
```

Comment:
As with `self.interface.enabled` - the "if" clause here shouldn't be required if you're using the interface's parsed value. If you use the interface value, you're guaranteed to have a parsed, valid value and don't need any additional validation - just the code to convert to an implementation-specific representation.

Reply:
I guess this is required because these functions get [called](https://github.com/pybee/toga/blob/9e887767f53d19ed45932678433c17c047f5e08e/src/gtk/toga_gtk/widgets/base.py#L13) even when the user won't explicitly set a `style` while making his widget - 
```
Traceback (most recent call last):
  File "/home/yugesh/Documents/beeware/toga/src/gtk/toga_gtk/app.py", line 140, in startup
    self.interface.startup()
  File "/home/yugesh/Documents/beeware/toga/src/core/toga/app.py", line 156, in startup
    self.main_window.content = self._startup_method(self)
  File "/home/yugesh/Documents/beeware/ctf.py", line 27, in build
    button = toga.Button('Calculate', on_press=calculate, enabled=False)
  File "/home/yugesh/Documents/beeware/toga/src/core/toga/widgets/button.py", line 24, in __init__
    self._impl = self.factory.Button(interface=self)
  File "/home/yugesh/Documents/beeware/toga/src/gtk/toga_gtk/widgets/base.py", line 13, in __init__
    self.interface.style.reapply()
  File "/home/yugesh/Documents/beeware/venv/lib/python3.5/site-packages/travertino/declaration.py", line 88, in reapply
    self.apply(style, getattr(self, style))
  File "/home/yugesh/Documents/beeware/toga/src/core/toga/style/pack.py", line 86, in apply
    self._applicator.set_background_color(value)
  File "/home/yugesh/Documents/beeware/toga/src/core/toga/style/applicator.py", line 30, in set_background_color
    self.widget._impl.set_background_color(color)
  File "/home/yugesh/Documents/beeware/toga/src/gtk/toga_gtk/widgets/button.py", line 23, in set_background_color
    self.native.override_background_color(value)
TypeError: Argument 1 does not allow None as a value
```

Reply:
With `self.interface.enabled` the advantage is that it is a parameter specified in the core constructor so it *has* to be **True** or **False**.